src/utils/errorid-generator.ts

```diff
@@ -0,0 +1,36 @@
+export class ErrorIdGenerator {
+  private generatedIds: Set<number> = new Set<number>();
+  private readonly idLength: number = 6;
+
+  generateUniqueErrorId(): number {
+    let newId: number;
+    do {
+      newId = this.generateNonZeroId();
+    } while (this.generatedIds.has(newId));
+
+    this.generatedIds.add(newId);
+    return newId;
+  }
+
+  private generateNonZeroId(): number {
+    let result = '';
+    for (let i = 0; i < this.idLength; i++) {
+      result += Math.floor(Math.random() * 9) + 1; // Generate 1-9
```

Comment:
Use cryptographically safe random number generation like: https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues

Reply:
I think this is not a risk here because the code using the generated value not requires it to be unpredictable. Its not opening an attack vector here.